tests/Unleash.Tests/Internal/Dependent_Features_Tests.cs

```diff
@@ -124,12 +124,12 @@ public void Depends_On_One_Enabled_Parent_Counts_Metrics_Only_For_Child()
             var result = unleash.IsEnabled("child-1");
 
             // Assert
-            unleash.services.MetricsBucket.StopCollectingMetrics(out var bucket);
-            var childMetrics = bucket.Toggles.Single(t => t.Key == "child-1").Value;
-            childMetrics.No.Should().Be(0L);
-            childMetrics.Yes.Should().Be(1L);
+            var bucket = unleash.services.engine.GetMetrics();
+            var childMetrics = bucket?.Toggles.Single(t => t.Key == "child-1").Value;
+            childMetrics?.No.Should().Be(0L);
```

Comment:
Oh that's very subtle, can we assert not null here? My eyes are telling me (perhaps wrong) that the asserts won't run if metrics is null

Reply:
Yeah +1 on that

Reply:
Do you mean something like this? https://github.com/Unleash/unleash-client-dotnet/pull/224/commits/e7752b4284febb6d026b619f3571092a335004d1

Unfortunately my editor type checker doesn't care about the assertion and still complains that `bucket` could possibly be null, but that's probably okay.

![image](https://github.com/Unleash/unleash-client-dotnet/assets/14320932/bd23dd54-5172-492b-9bcc-28eba96a2463)

Reply:
```
bucket.Should().NotBeNull();
```
Yuh. Happy

Reply:
Yes, happy!